staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy_test.go

```diff
@@ -313,6 +348,10 @@ func TestProxyHandler(t *testing.T) {
 				t.Errorf("%s: expected %v, got %v", name, e, a)
 				return
 			}
+			if e, a := targetServer.Listener.Addr().String(), target.host; tc.expectedCalled && !reflect.DeepEqual(e, a) {
```

Comment:
does it work with the test case "service unresolveable"? 
In that test case, serviceResolver: &mockedRouter{err: fmt.Errorf("unresolveable")},

Reply:
it works for me, are there any problems on your side?

Reply:
i see why it would work, because tc.expectedCalled is false for the test case "service unresolveable", hence !reflect.DeepEqual(e, a) would not be called at all.